go/runtime/client/submitter.go

```diff
@@ -71,6 +75,17 @@ func (w *txSubmitter) checkBlock(blk *block.Block) error {
 		return nil
 	}
 
+	// If we have local storage, wait for it to sync up to this point.
+	if w.storageCommitteeNode != nil {
+		select {
+		case <-w.storageCommitteeNode.WaitForRound(blk.Header.Round):
```

Comment:
I think this would be more general if it was handled in the storage backend wrapper exposed by the storage node (e.g., used with `RegisterStorage`) instead of directly here. Because there can be other uses of storage in the client when it is also a storage node.

E.g., create a simple wrapper in storage worker's `registerRuntime` that basically makes it wait for rounds for all operations (note only the backend that is wrapped is the one passed to `RegisterStorage`).